sdk/src/azure/iot/az_iot_pnp_client_commands.c

```diff
@@ -0,0 +1,83 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// SPDX-License-Identifier: MIT
+
+#include <stdint.h>
+
+#include <azure/core/az_precondition.h>
+#include <azure/core/az_result.h>
+#include <azure/core/az_span.h>
+#include <azure/core/internal/az_precondition_internal.h>
+#include <azure/iot/az_iot_hub_client.h>
+#include <azure/iot/az_iot_pnp_client.h>
+
+#include <azure/core/internal/az_log_internal.h>
+#include <azure/core/internal/az_precondition_internal.h>
+
+#include <azure/core/_az_cfg.h>
+
+static const az_span commands_topic_prefix = AZ_SPAN_LITERAL_FROM_STR("$iothub/methods/");
+static const az_span commands_topic_filter_suffix = AZ_SPAN_LITERAL_FROM_STR("POST/");
+static const az_span commands_response_topic_properties = AZ_SPAN_LITERAL_FROM_STR("/?$rid=");
+static const az_span command_separator = AZ_SPAN_LITERAL_FROM_STR("*");
+
+AZ_NODISCARD az_result az_iot_pnp_client_commands_parse_received_topic(
+    az_iot_pnp_client const* client,
+    az_span received_topic,
+    az_iot_pnp_client_command_request* out_request)
+{
+  _az_PRECONDITION_NOT_NULL(client);
```

Comment:
Add a precondition that the client is initialized (i.e. hostname is set to something non-empty).